tests/ndarrays/test_ndarrays.c

```diff
@@ -0,0 +1,278 @@
+#include "ndarrays.h"
+#include <unistd.h>
+
+
+#define m_assert(X, Y, dscr) my_assert(X , Y, dscr, __func__, __FILE__, __LINE__)
+
+#define my_assert(X , Y, dscr, __func__, __FILE__, __LINE__) _Generic((X), double: assert_double,\
+							float: assert_float,\
+							int: assert_int,\
+							double complex : assert_complex_double,\
+							default: assert_ns)(X , Y, dscr, __func__, __FILE__, __LINE__)
+#define getname(X) #X
+
+void assert_double(double v1 , double v2, const char *dscr, const char * func, const char *file, int line)
+{
+
+	if (v1 != v2)
+	{
+		printf("[FAIL] %s:%d:%s\n", file, line, func);
+		printf("[INFO] %s:%f != %s:%f\n", getname(v1), v1, getname(v2),v2);

```

Comment:
Does `getname(v1)` just print `v1`? Why is the `getname` function helpful here?

Reply:
i just wanted to have the variable name and the value next to each other e.g:  `INFO : get_index(x, 3, 1):25 != c_index:26`
i did fixed it to work how i wanted